src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -971,12 +967,12 @@ private BoundExpression VisitUnaryOperator(BoundUnaryOperator node)
 
         private BoundExpression ExprFactory(string name, params BoundExpression[] arguments)
         {
-            return _bound.StaticCall(_binderFlags, ExpressionType, name, arguments);
+            return _bound.StaticCall(ExpressionType, name, arguments);
         }
 
         private BoundExpression ExprFactory(string name, ImmutableArray<TypeSymbol> typeArgs, params BoundExpression[] arguments)
         {
-            return _bound.StaticCall(_binderFlags, ExpressionType, name, typeArgs, arguments);
+            return _bound.StaticCall(_allowNonPublicTypeArguments ? BinderFlags.IgnoreAccessibility : BinderFlags.None, ExpressionType, name, typeArgs, arguments);
```

Comment:
Setting IgnoreAccessibility is stronger than allowing non-public type arguments.